components/ParticipantFormUpdate.tsx

```diff
@@ -29,12 +30,27 @@ export default function ParticipantFormUpdate({
 
   const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    await ParticipantService(eventId).update(participantData.id, data);
-    setIsSubmitted(true);
-    setTimeout(() => {
-      setIsSubmitted(false);
-      router.push(`/events/${eventId}`);
-    }, 3000);
+    const path = `/events/${eventId}`;
+    const timeout = 3000;
+    try {
+      const res = await ParticipantService(eventId).update(
+        participantData.id,
+        data,
+      );
+      console.log(res);
+      setIsSubmitted(true);
+      await redirectAfterTimeout(timeout, path);
+    } catch (error) {
+      setIsFailed(true);
+      await redirectAfterTimeout(timeout, path);
+    }
+  };
+
+  const redirectAfterTimeout = async (timeout: number, path: string) => {
+    await new Promise((resolve) => setTimeout(resolve, timeout));
+    setIsFailed(false);
+    setIsSubmitted(false);
+    router.push(path);
```

Comment:
I guess this can be a bit simplified:

```suggestion
    await new Promise((resolve) => {
      setTimeout(() => {
            setIsFailed(false);
            setIsSubmitted(false);
            router.push(path);
      }, timeout);
    });
```

Reply:
yes you are right. But I've simplified it even further. There is no need to have this logic in its own function, because the redirect should be always at the end.
Maybe on a later stage we can redirect to the forms again, on an error, but as of right now, this is the current behaviour.